test/functional/tests/fault_injection/test_fault_injection_with_mounted_core.py

```diff
@@ -0,0 +1,169 @@
+def test_stop_cache_with_mounted_partition(cache_mode):
+    """
+        title: Fault injection test for stopping the OpenCAS with mounted partition.
+        description: |
+          False positive test of the ability to stop while partition
+          is still mounted on OpenCAS device.
+        pass_criteria:
+          - No system crash while loading cache.
+          - Unable to stop CAS device.
+          - Unable to remove module when partition is mounted.
+    """
+    with TestRun.step("Prepare cache and core devices. Start OpenCAS."):
+        cache_dev = TestRun.disks['cache']
+        cache_dev.create_partitions([Size(1, Unit.GibiByte)])
+        cache_dev = cache_dev.partitions[0]
+        core_dev = TestRun.disks['core']
+        core_dev.create_partitions([Size(4, Unit.GibiByte)])
+        core_dev = core_dev.partitions[0]
+        cache = casadm.start_cache(cache_dev, cache_mode, force=True)
+
+    with TestRun.step("Add core device with xfs filesystem and mount it."):
+        core_dev.create_filesystem(Filesystem.xfs)
+        core = cache.add_core(core_dev)
+        core.mount(mount_point)
+
+    with TestRun.step("Try to remove core from cache."):
+        output = TestRun.executor.run_expect_fail(cli.remove_core_cmd(cache_id=str(cache.cache_id),
+                                                                      core_id=str(core.core_id)))
+        if not output.stderr:
+            TestRun.fail("Core remove succeeded (should fail)!")
+
+    with TestRun.step("Try to stop OpenCAS."):
+        output = TestRun.executor.run_expect_fail(cli.stop_cmd(cache_id=str(cache.cache_id)))
+        if not output.stderr:
+            TestRun.fail("OpenCAS stop succeeded (should fail)!")
+
+    with TestRun.step("Unmount core device."):
+        core.unmount()
+
+    with TestRun.step("Stop all caches."):
+        casadm.stop_all_caches()
+
+
+@pytest.mark.parametrize("cache_mode", CacheMode)
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_add_occupied_core(cache_mode):
+    """
+        title: Fault injection test to adding already used core to the OpenCAS.
+        description: |
+          False positive test of the ability to add core to OpenCAS
+          while its already used by the another OpenCAS instance.
+        pass_criteria:
+          - Adding the core from first cache to the second one fails.
+    """
+    with TestRun.step("Prepare two caches and one core devices."):
+        cache_dev = TestRun.disks['cache']
+        cache_dev.create_partitions([Size(2, Unit.GibiByte), Size(2, Unit.GibiByte)])
+        cache_dev1 = cache_dev.partitions[0]
+        cache_dev2 = cache_dev.partitions[1]
+        core_dev = TestRun.disks['core']
+        core_dev.create_partitions([Size(4, Unit.GibiByte)])
+        core_dev = core_dev.partitions[0]
+
+    with TestRun.step("Start first OpenCAS instance"):
```

Comment:
OpenCAS -> cache
There is one instance of CAS in the system, and that instance can start many cache instances